src/components/vm/overview/watchdog.tsx

```diff
@@ -97,39 +97,46 @@ export const WatchdogModal = ({
 
     const Dialogs = useDialogs();
 
-    const setWatchdogColdplug = () => {
-        return domainSetWatchdog({
-            connectionName: vm.connectionName,
-            vmName: vm.name,
-            action: watchdogAction,
-            defineOffline: true,
-            hotplug: false,
-            isWatchdogAttached,
-        })
-                .then(Dialogs.close, exc => setDialogError({ text: _("Failed to configure watchdog"), detail: exc.message, variant: "danger" }));
+    async function setWatchdog(vm: VM, hotplug: boolean) {
+        if (isWatchdogAttached) {
+            await virtXmlEdit(vm, "watchdog", 1, { action: watchdogAction });
+        } else {
+            await virtXmlAdd(
+                vm,
+                "watchdog",
+                { action: watchdogAction },
+                { update: hotplug, "no-define": hotplug },
+            );
+        }
+    }
+
+    const setWatchdogColdplug = async () => {
+        try {
+            await setWatchdog(vm, false);
+            Dialogs.close();
+        } catch (exc) {
+            setDialogError({ text: _("Failed to configure watchdog"), detail: String(exc), variant: "danger" });
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2334-68be91e9-20250929-092324-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)